### DISCUSSION TITLE
Accelerate community engagement with TorchSharp - suggestion

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [xamarin/TorchSharp#395] [Ideas]
The motivation for TorchSharp to [support porting of code from python to .NET is obvious](https://github.com/dotnet/machinelearning/issues/5918#issuecomment-918537115). 

To get more users to try up TorchSharp using the existing implemented work, the **TorchSharp Unit Tests** is a suitable first start.

Since the motivation assumes that there is abundant of books, learning videos and courses on PyTorch, what this repository can support is to provide e.g. either a Wiki or [Documents ](https://github.com/dotnet/TorchSharp/tree/main/docfx)to organize these unit tests according to the standards way how books, courses would introduce PyTorch.

For an example - start with organizing the unit tests in [TestTorchTensor.cs](https://github.com/dotnet/TorchSharp/blob/main/test/TorchSharpTest/TestTorchTensor.cs) into the following groupings:

# Tensors
### Automatic Differentiation (Autograd)

Ref: Chapter 2 of PyTorch Pocket Reference (O Reilly 2021)

Hopefully, with time, there will be more new users or existing users to join the process and use these tentative groupings to feedback, contribute, refine and organize these unit tests to jumpstart TorchSharp documentation for new users.

**Comment by GeorgeS2019** [comment]
New member please introduce yourself [here](https://github.com/dotnet/TorchSharp/discussions/333#discussioncomment-1607347).

**Comment by GeorgeS2019** [comment]
@dayo05 what do you think of adding **topic** structure to TorchSharp Unit Tests? Especially  [TestTorchTensor.cs](https://github.com/dotnet/TorchSharp/blob/main/test/TorchSharpTest/TestTorchTensor.cs)

**Reply by dayo05** [reply]
But I think making wiki is better for others who naver use pytorch, etc.
